diesel/src/expression_methods/bool_expression_methods.rs

```diff
@@ -37,17 +37,14 @@ pub trait BoolExpressionMethods: Expression + Sized {
     /// assert_eq!(expected, data);
     /// #     Ok(())
     /// # }
-    fn and<T>(self, other: T) -> dsl::And<Self, T>
+    fn and<T, ST>(self, other: T) -> dsl::MaybeNullableAnd<Self, T, ST>
```

Comment:
@weiznich How would you write this if not creating such another wrapper? (https://github.com/diesel-rs/diesel/issues/2589#issuecomment-748304471)